src/site/content/en/reliable/codelab-reliability-precaching/index.md

```diff
@@ -0,0 +1,272 @@
+    ```
+
+    If you don't see `service-worker.js` in the `dist/` directory, run the
+    following command in the Glitch console to refresh the visual filesystem:
+
+    ```
+    refresh
+    ```
+
+1. Open `dist/service-worker.js`. You should see that a precache
+   manifest has been injected into your service worker, with the remainder
+   of the service worker left untouched:
+
+    ```js
+    // ...
+    workbox.precaching.precacheAndRoute([
+      {
+        "url": "app.js",
+        "revision": "225a3b50beff884b6b0382367cd39da2"
+      },
+      {
+        "url": "index.html",
+        "revision": "c267d5f79131f190d2b157eb9c98137a"
+      },
+      {
+        "url": "offline.partial.html",
+        "revision": "46c20898f3d0f9c78b56834287a47d4a"
+      },
+      {
+        "url": "styles.css",
+        "revision": "aab5b6d2737656fcfd813373178afd10"
+      }
+    ]);
+    // ...
+    ```
+
+The injected array includes all files that the service worker will precache.
+Each object in the array contains a `revision` attribute, which Workbox uses
+to track and update precached resources that have been modified. If a
+URL-revision combination in the manifest changes, the service worker knows that
+the previous cached entry is no longer valid and needs to be updated.
+
+To see how Workbox's versioning works, copy the `revision` attribute for
+`app.js` so you can reference it later. Then add a comment anywhere in the
+`app.js` file and run Workbox again from the Glitch console:
+
+```
+workbox injectManifest workbox-config.js && refresh
+```
+
+Look at `dist/service-worker.js` to verify that the `revision` attribute for
+`app.js` has changed, similar to the example below:
+
+```js/5/4
+// ...
+workbox.precaching.precacheAndRoute([
+  {
+    "url": "app.js",
+    "revision": "225a3b50beff884b6b0382367cd39da2"
+    "revision": "879b834765f0c7c7ce4b72f877b29ad6"
+  },
+  // ...
+]);
+// ...
+```
+
+Now you're ready to see precaching in action. Modify `server.js` to serve the
+generated service worker file from the `dist/` directory (instead of the
+`src/` directory):
+
+```js/3/2
+// route for service worker
+app.get(/service-worker\.js/, function(request, response) {
+  response.sendFile(__dirname + `/src/service-worker.js`);
+  response.sendFile(__dirname + `/dist/service-worker.js`);
+});
+```
+
+Navigate to the app and refresh the page. In the __Network__ tab of DevTools,
+you should see the service worker fetch and precache the app shell files in the
+precache manifest.
+
+<figure class="w-figure w-figure--center">
+  <img class="w-screenshot" src="./precached-files.png" alt="A screenshot of
+  Chrome DevTools showing the precached files.">
+</figure>
+
+{% Aside %}
+Workbox appends a special \_\_WB_REVISION__ URL query parameter to URLs that
+need revision information. If a URL already contains version information (e.g.,
+a hash value), Workbox's build tools will detect it and exclude a revision
```

Comment:
There's a little nuance here; you have to specifically configure the build tools to detect that.

Can you link out tot https://developers.google.com/web/tools/workbox/modules/workbox-cli#common-dontCacheBustURLsMatching to provide that context?